### PR TITLE
fix mouse right click selection

### LINKED ISSUE #10066: right clicking doesn't select the right character
### Summary

trying to select a character with a right click actually selects a character in the wrong position and not the one under the mouse cursor

looking at the source it seems that when right clicking the event handler is using gutter coordinates instead of document coordinates

### Reproduction Steps

 

I tried this:

1. open any file
2. right click with the mouse to select a character

I expected this to happen:
the character under the mouse cursor is selected

Instead, this happened:
a character few cells away is selected

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Linux

### Terminal Emulator

Alacritty, Xterm

### Installation Method

pacman

### Helix Version

helix 23.10

### PR DESCRIPTION
**Pull request by diegostafa**
fix #10066

### CODE REVIEW

File: helix-term/src/ui/editor.rs [hunk 1]
```diff
@@ -1231,22 +1231,19 @@ impl EditorView {
...
-                    if let Some(pos) =
-                        view.pos_at_visual_coords(doc, coords.row as u16, coords.col as u16, true)
-                    {
-                        doc.set_selection(view_id, Selection::point(pos));
-                        if modifiers == KeyModifiers::ALT {
-                            commands::MappableCommand::dap_edit_log.execute(cxt);
-                        } else {
-                            commands::MappableCommand::dap_edit_condition.execute(cxt);
-                        }
+                    let doc = current!(cxt.editor).1;
```
Comment (the-mikedavis) [inline helix-term/src/ui/editor.rs:1237]:
```suggestion
                    let doc = doc_mut!(cxt.editor);
```

**Review by the-mikedavis — APPROVED** [review]

File: helix-term/src/ui/editor.rs [hunk 2]
```diff
@@ -1231,22 +1231,19 @@ impl EditorView {
...
+                    doc.set_selection(view_id, Selection::point(pos));
 
-                        return EventResult::Consumed(None);
+                    if modifiers == KeyModifiers::ALT {
+                        commands::MappableCommand::dap_edit_log.execute(cxt);
+                    } else {
+                        commands::MappableCommand::dap_edit_condition.execute(cxt);
                     }
+
+                    return EventResult::Consumed(None);
```
Comment (pascalkuthe) [inline helix-term/src/ui/editor.rs:1246]: This is missing ane`ensure_cursor_in_view` call (see what the left click is doing above)

**Comment by archseer** [comment]
I think this was somewhat deliberate, we're editing breakpoints which are only supported with column offsets (start of line) in many debuggers

**Comment by diegostafa** [comment]
> I think this was somewhat deliberate, we're editing breakpoints which are only supported with column offsets (start of line) in many debuggers

i changed it so that when clicking on the margin it edits the breakpoint, and when clicking inside the document it selects the character.
the main problem with the previous behaviour is that it just feels wrong to click somewhere and set the selection somewhere else

**Review by pascalkuthe — APPROVED** [review]

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**